spec/outputs/graphite_spec.rb

```diff
@@ -1,86 +1,49 @@
 require_relative '../spec_helper'
 
-describe LogStash::Outputs::Graphite do
-
-  let(:port) { 4939 }
-  let(:config) do <<-CONFIG
-     input {
-        generator {
-          message => "foo=fancy bar=42"
-          count => 1
-          type => "generator"
-        }
-      }
-
-      filter {
-        kv { }
-      }
-
-      output {
-        graphite {
-          host => "localhost"
-          port => #{port}
-          metrics => [ "hurray.%{foo}", "%{bar}" ]
-        }
-      }
-  CONFIG
+class LogStash::Outputs::Graphite
+  def socket
+    @socket
   end
+  def connect
+    @socket = Mocks::Server.new
```

Comment:
I tend to avoid using DI as reopening classes is (for me) a cleaner (and the Ruby) way to redefine a class' behavior. It comes down to a matter of taste, I guess.